modules/tensor_mechanics/src/materials/abaqus/AbaqusUMATStress.C

```diff
@@ -204,11 +214,36 @@ AbaqusUMATStress::computeQpStress()
   const Real * myDFGRD1 = &(FBar_fortran(0, 0));
   const Real * myDROT = &(DROT_fortran(0, 0));
 
+  // More local copies of materials so we can (optionally) rotate
+  RankTwoTensor stress_old = _stress_old[_qp];
+  RankTwoTensor total_strain_old = _total_strain_old[_qp];
+  RankTwoTensor strain_increment = _strain_increment[_qp];
+
+  // check if we need to rotate to intermediate configuration
+  if (_use_orientation)
+  {
+    // keep track of total rotation
+    _total_rotation[_qp] = _rotation_increment[_qp] * _total_rotation_old[_qp];
+    // rotate stress/strain/increment from reference configuration to intermediate configuration
+    stress_old.rotate(_total_rotation_old[_qp].transpose());
+    total_strain_old.rotate(_total_rotation_old[_qp].transpose());
+
+    if (_decomposition_method == ComputeFiniteStrain::DecompMethod::HughesWinget)
+      strain_increment.rotate(_total_rotation[_qp].transpose());
+    else
+      strain_increment.rotate(_total_rotation_old[_qp].transpose());
+  }
+  else if (_decomposition_method == ComputeFiniteStrain::DecompMethod::HughesWinget)
+    // rotate old stress to reference configuration
+    stress_old.rotate(_rotation_increment[_qp]);
+
   // copy because UMAT does not guarantee constness
   for (const auto i : make_range(9))
   {
     _aqDFGRD0[i] = myDFGRD0[i];
     _aqDFGRD1[i] = myDFGRD1[i];
+    // TODO: Use old rotation increment for DROT if Rashid so UMAT state variables
+    // can be rotated to the same reference frame as stress (ie. old stress)
```

Comment:
Why do we need to do something different here?

Reply:
> Can we have a constant rotation, defining the initial anisotropy, that gets composed with rotation increments?

I'm not sure I understand. My goal was to make this behave consistent with _Abaqus_ (for better or worse). There is also a subtlety in that Abaqus will carry along the intermediate configuration for any specified orientation, including a 0-degree rotation. Hence the different procedure with and without orientation.

> Why do we need to do something different here?

UMAT state variables do not get updated due to rotations. The idea here is to pass the appropriate rotation increment to the UMAT so the material model can update its own _tensor-based_ state variables manually. I just don't have a good test case for this.